src/Microsoft.DotNet.VersionTools/BuildManifest/BuildManifestLocation.cs

```diff
@@ -0,0 +1,43 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.DotNet.VersionTools.Automation;
+using System;
+
+namespace Microsoft.DotNet.VersionTools.BuildManifest
+{
+    public class BuildManifestLocation
+    {
+        public GitHubProject GitHubProject { get; }
+
+        public string GitHubRef { get; }
+
+        public string GitHubBasePath { get; }
+
+        public BuildManifestLocation(
+            GitHubProject gitHubProject,
+            string gitHubRef,
+            string gitHubBasePath)
+        {
+            if (gitHubProject == null)
+            {
+                throw new ArgumentNullException(nameof(gitHubProject));
+            }
+
+            if (string.IsNullOrEmpty(gitHubRef))
+            {
+                throw new ArgumentException(nameof(gitHubRef));
+            }
+
+            if (gitHubBasePath == null)
```

Comment:
Should this check be `string.IsNullOrEmpty`?

Reply:
Originally I thought empty is fine because it means putting a manifest in the root dir. But since the base path doesn't include the trailing `/`, I would have had to handle empty through the rest of the code, which I haven't done. Changing to `IsNullOrEmpty`--root isn't a useful scenario right now.